### DISCUSSION TITLE
关于Markup Schema JSON Schema 和jsx这三种模式

### DISCUSSION THREAD

**Discussion by kkodeok** [alibaba/formily#1379] [Q&A 2.x] [answered]
请问Markup Schema JSON Schema 和jsx这三种模式是为了什么产生了三种写法呢  ，以及三种写法的对功能和实现上的区别是？

**Comment by janryWang** [comment, marked as answer]
第一，纯JSX模式是一种很轻量很灵活很自由的模式，类比React Hook Form那种开发模式，适用于简单场景，或者扩展自定义组件内部的子表单实现
第二，JSON Schema模式，适用于表单搭建场景
第三，Markup Schema模式，适用于复杂表单场景，因为借助协议驱动能力，我们可以封装抽象很多场景组件，比如FormStep/FormTabs这些组件，在纯JSX模式则是很难做到的

**Reply by kkodeok** [reply]
谢谢解答 但是还是对MarkUp-Schema不是很理解 之前没有接触过这个结构，查询一些资料后了解到也是跟json schema一样是一种协议表达方式，不知道理解是否正确以及跟json-schema的区别是在？

**Reply by janryWang** [reply]
markup schema跟json schema等价，只是在jsx中可以拥有更好的智能提示，可维护性更好

**Reply by kkodeok** [reply]
好的明白了  非常感谢解答

**Reply by kkodeok** [reply]
另外想补充问一下  这三种在实际项目中是可以混用的吗

**Reply by janryWang** [reply]
可以的

**Reply by kkodeok** [reply]
好的 谢谢
